### PR TITLE
source-sigma-computing contribution from aazam-gh

### LINKED ISSUE #46625: [connector-request] Sigma Computing Source
This is a **suggestion** to create a connector for **Sigma Computing Source** using the UI Connector Builder.
The website for the connector is https://www.sigmacomputing.com.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by aazam-gh**
## What

This PR adds a new connector for source Sigma Computing (source-sigma-computing).

Contributor-provided connector description:

This is the setup for the Sigma Computing source that ingests data from the sigma API. 

Sigma is next-generation analytics and business intelligence that scales billions of records using spreadsheets, SQL, Python, or AI—without compromising speed and security https://www.sigmacomputing.com/

In order to use this source, you must first create an account on Sigma Computing. Go to Account &gt; General Settings and review the Site section for the Cloud provider, this will be used to find the base url of your API. Compare it at https://help.sigmacomputing.com/reference/get-started-sigma-api

Next, head over to Developer Access and click on create. This will generate your Client ID and Client Secret required by the API. You can learn more about the API here https://help.sigmacomputing.com/reference

## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `client_id` | `string` | Client ID.  |  |
| `client_secret` | `string` | Client secret.  |  |
| `client_refresh_token` | `string` | Refresh token.  |  |
| `oauth_access_token` | `string` | Access token. The current access token. This field might be overridden by the connector based on the token refresh endpoint response. |  |
| `oauth_token_expiry_date` | `string` | Token expiry date. The date the current access token expires in. This field might be overridden by the connector based on the token refresh endpoint response. |  |
| `base_url` | `string` | Base URL. The base url of your sigma organization |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| files | id | DefaultPaginator | ✅ |  ❌  |
| connections | connectionId | DefaultPaginator | ✅ |  ❌  |
| datasets | datasetId | DefaultPaginator | ✅ |  ❌  |
| members | memberId | DefaultPaginator | ✅ |  ❌  |
| teams | teamId | DefaultPaginator | ✅ |  ❌  |
| templates | templateId | DefaultPaginator | ✅ |  ❌  |
| workspaces |  | DefaultPaginator | ✅ |  ❌  |
| workbooks | workbookId | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

**Comment by aazam-gh** [comment]
@marcosmarxm As sigma has a trial of only 7 days. Would be great if you can review this with priority!

File: airbyte-integrations/connectors/source-sigma-computing/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,909 @@
...
+          page_token_option:
+            type: RequestOption
+            inject_into: request_parameter
+            field_name: page
+          page_size_option:
+            type: RequestOption
+            field_name: limit
+            inject_into: request_parameter
+          pagination_strategy:
+            type: OffsetIncrement
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-sigma-computing/manifest.yaml:60]: I don't believe the offset is the correct pagination method for the API. https://help.sigmacomputing.com/reference/listdatasets It uses pages, can you double check it?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-sigma-computing/manifest.yaml (hunk 1 above)
Comment (aazam-gh) [reply airbyte-integrations/connectors/source-sigma-computing/manifest.yaml]: You 
are correct. Updated :)

**Review by marcosmarxm — APPROVED** [review]
Great contribution @aazam-gh

**Pull request merged**